File: MIssing Numbers 24-Aprill/Solution.py
def gcd(a,b):
    if b == 0:
        return a
    else:
        return gcd(b,a%b)
def minimumNumber(n,arr):
    ans = 0
    for i in range(len(arr)):
        ans = gcd(ans,arr[i])
    return ans 

list = [3,2,4]
n = 3
print(minimumNumber(n,list))

File: MIssing Numbers 24-Aprill/Solution.java
class Solution {
    public static int gcd(int a,int b){
        if(b == 0) return a;
        else return gcd(b,a%b);
    }
    public static int minimumNumber(int n, int[] arr) {
        // code here
        int ans = 0;
        for(int i = 0 ; i < n ; i++){
            ans = gcd(ans,arr[i]);
        }
        return ans;
    }
    public static void main(String[] args) {
        int[] arr = {2,4,6,5,4};
        int n = 5;
        System.out.println(minimumNumber(n, arr));
    }
    
}